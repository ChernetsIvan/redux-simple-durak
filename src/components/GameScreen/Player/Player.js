import React from "react";
import PropTypes from "prop-types";

import InvisibleCard from "./../Card/InvisibleCard";
import Card from "./../Card/Card";

//Классы-контейнеры кнопок
import PlayerTakeButtonContainer from "./../../../containers/GameScreen/Player/Buttons/PlayerTakeButtonContainer";
import AiTakeButtonContainer from "./../../../containers/GameScreen/Player/Buttons/AiTakeButtonContainer";
import RemoveCardsButtonContainer from "./../../../containers/GameScreen/Player/Buttons/RemoveCardsButtonContainer";

class Player extends React.Component {
  constructor(props) {
    super(props);

    this.onPrevClick = this.onPrevClick.bind(this);
    this.onNextClick = this.onNextClick.bind(this);

    this.state = {
      startInd: 0,
      endInd: 9
    };
  }

  onPrevClick() {
    this.setState((prevState)=>({
      startInd: prevState.startInd - 1,
      endInd: prevState.endInd - 1
    }));
  }

  onNextClick() {
    this.setState((prevState)=>({
      startInd: prevState.startInd + 1,
      endInd: prevState.endInd + 1
    }));
  }

  render() {
    let cards = this.props.cards.map(el => {
      return (
        <Card
          id={el.id}
          key={el.id}
          rank={el.rank}
          suit={el.suit}
          onCardClick={this.props.onCardClick}
          bootStrapColClass="col-1"
          hidden={false}
        />
      );
    });

    if (cards.length < 10) {
      this.setState({
        startInd: 0,
        endInd: 9
      });
    }

    let output = null;
    if (this.props.cards.length > 0) {
      //Показ кнопок Prev и Next
      let out_PrevButton = null;
      if (this.state.startInd > 0 && this.props.cards.length > 10) {
        out_PrevButton = (
          <button onClick={this.onPrevClick} className="btn btn-info">
            <b>&larr;{this.state.startInd}</b>
          </button>
        );
      }

      let out_NextButton = null;
      if (
        this.state.endInd < this.props.cards.length - 1 &&
        this.props.cards.length > 10
      ) {
        out_NextButton = (
          <button onClick={this.onNextClick} className="btn btn-info">
            <b>{this.props.cards.length - 1 - this.state.endInd}&rarr;</b>
          </button>
        );
      }

      //Выводим максимум 10 карт: от startInd до endInd
      let outCards = [];
      if (this.props.cards.length > 10) {
        outCards = cards.slice(
          this.state.startInd,
          this.state.endInd + 1
        );
      } else {
        outCards = cards;
      }

      output = (
        <div className="container">
          <div className="row justify-content-center">
            <div className="col-1 align-self-center">{out_PrevButton}</div>
            {outCards}
            <div className="col-1 align-self-center">{out_NextButton}</div>
          </div>
        </div>
      );
    } else {
      output = <InvisibleCard />;
    }
    return (
      <div className="container">
        <div className="row">
          <div className="col-1" />
          <div className="col-10">{output}</div>
          <div className="col-1" />
        </div>
        <div className="row justify-content-center">
          <div className="col-auto">
            <RemoveCardsButtonContainer />
            <AiTakeButtonContainer />
            <PlayerTakeButtonContainer />
          </div>
        </div>
      </div>
    );
  }
}

export default Player;

Player.propTypes = {
  cards: PropTypes.array.isRequired,
  onCardClick: PropTypes.func.isRequired
};
