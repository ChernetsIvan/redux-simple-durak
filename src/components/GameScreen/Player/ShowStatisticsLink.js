import React from "react";
import PropTypes from "prop-types";
import { Link } from "react-router";

import * as gameModes from "./../../../constants/GameModes";

const ShowStatisticsLink = ({ gameMode }) => {
  let dash = <span>&mdash;</span>;
  return gameMode === gameModes.AIWin ||
  gameMode === gameModes.PlayerWin ||
  gameMode === gameModes.TheDraw ? (
    <Link
      to="/statistics"
      className="btn btn-secondary ml-2"
    >
      {dash} Игра закончена, смотреть статистику!
    </Link>
  ) : null;
};

export default ShowStatisticsLink;

ShowStatisticsLink.propTypes = {
  gameMode: PropTypes.string.isRequired
};
