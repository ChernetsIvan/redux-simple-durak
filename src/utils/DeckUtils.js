import { pull, find } from 'lodash-es';

import {cardRanks, cardSuits} from './../constants/GameConstants';

import CardModel from './../models/CardModel';

class DeckUtils{
    //отсортировать входной массив карт по возрастанию isAsc=true
    //(убыванию-> isAsc=false)  по параметру 'power'
    static sortInputDeckByPower(inputDeck, isAsc){
        inputDeck.sort(function(a,b){
            if(isAsc){
                return a.power - b.power;
            }else{
                return b.power - a.power;
            }        
        });
    }

    //Возвращает наполненную всеми 36-ю картами колоду
    static fillCards(trumpSuit){
        let fullDeck = [];
        //Устанавливает power для каждой карты по принципу:
        //козырь->10*rank.cardValue, НЕкозырь->1*rank.cardValue
        let z = 0;
        let power, coef;               
        cardSuits.forEach(function(element,index, array){
            if(element.suit === trumpSuit.suit){
                coef = 10;
            }else{
                coef = 1;
            }
            cardRanks.forEach(function(element2, index2, array2){
                power = coef * element2.cardValue;
                fullDeck.push(
                    new CardModel(z, element2, element, power)
                );
                z++;
            });
        });
        return fullDeck;
    }

    //Возвращает перемешанную колоду
    static shuffle(inputDeck){
        var newFullDeck = [];
        while(inputDeck.length > 0){
            //Случайный индекс в inputDeck
            let randIndex = Math.random() * (inputDeck.length-1);
            randIndex = Math.round(randIndex);

            let card = inputDeck[randIndex];
            newFullDeck.push(card);

            inputDeck.splice(randIndex, 1);
        }
        return newFullDeck;
    }

    //Перемещает первую попавшуюся карту с козырной мастью
    //в конец колоды 
    static moveAnyCardWithTrumpSuitToTailOfFullDeck(trumpSuit, fullDeck){
        let card = find(fullDeck, (card) => {
            return card.suit.suit === trumpSuit.suit;
        });
        pull(fullDeck, card);
        fullDeck.push(card);
    }

    //взять из НАЧАЛА (!) колоды fullDeck и положить в inputDeck столько карт, 
    //чтобы в inputDeck в итоге было 6 карт
    static giveUpToSixCards(fullDeck, inputDeck){
        //если хотя бы одно из условий не выполняется, в цикл не заходим:
        while(inputDeck.length < 6 && fullDeck.length > 0){
            let card = fullDeck.shift();
            inputDeck.push(card);
        }
    }

    //Возвращает козырную масть (не карту!) для будущей игры
    static chooseSuitForTrumpCardInFuture(){
        let randIndex = Math.random() * (cardSuits.length-1);
        randIndex = Math.round(randIndex);
        return cardSuits[randIndex];
    }
}

export default DeckUtils;