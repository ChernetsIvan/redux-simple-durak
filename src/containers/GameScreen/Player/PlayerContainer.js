import { connect } from 'react-redux';
import Player from './../../../components/GameScreen/Player/Player';
import {clickOnPrev, clickOnNext, clickOnCard} from './../../../actions';

const mapStateToProps = store => ({
    cards: store.game.playerCards,
    playerStartInd: store.game.playerStartInd,
    playerEndInd: store.game.playerEndInd    
});

const mapDispatchToProps =  dispatch => ({   
    onPrevClick: () => {
        dispatch(clickOnPrev());
    },
    onNextClick: () => {
        dispatch(clickOnNext());
    },
    onCardClick: (cardId) => {
        dispatch(clickOnCard(cardId));
    }
});

export default connect(mapStateToProps, mapDispatchToProps)(Player);