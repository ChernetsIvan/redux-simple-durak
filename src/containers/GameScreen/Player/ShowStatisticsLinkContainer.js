import { connect } from "react-redux";
import ShowStatisticsLink from "./../../../components/GameScreen/Player/ShowStatisticsLink";

import { clickOnGameOverLink } from "./../../../actions";

const mapStateToProps = store => ({
  gameMode: store.game.gameMode
});

const mapDispatchToProps = dispatch => ({
  onGameOverClick: gameMode => {
    dispatch(clickOnGameOverLink(gameMode));
  }
});

export default connect(mapStateToProps, mapDispatchToProps)(ShowStatisticsLink);
