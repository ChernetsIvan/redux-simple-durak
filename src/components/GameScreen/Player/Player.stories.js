import React from "react";

import { storiesOf } from "@storybook/react";
import { action } from "@storybook/addon-actions";

import { Provider } from "react-redux";
import { createStore } from "redux";

import Player from "./Player";
import DeckUtils from "./../../../utils/DeckUtils";
import * as gameModes from "./../../../constants/GameModes";

import reducer from "./../../../reducers";

const store = createStore(reducer);

const cards = DeckUtils.fillCards({ suit: "Ðš" });

const clickOnCard = action("Card was clicked!");

storiesOf("Player", module)
  .addDecorator(story => <Provider store={store}>{story()}</Provider>)
  .add("with 4 cards", () => (
    <Player cards={cards.slice(10, 14)} onCardClick={clickOnCard} />
  ))
  .add("with 7 cards", () => (
    <Player cards={cards.slice(28, 35)} onCardClick={clickOnCard} />
  ))
  .add("with 12 cards", () => (
    <Player cards={cards.slice(0, 12)} onCardClick={clickOnCard} />
  ))
  .add("with 36 cards", () => (
    <Player cards={cards.slice(0, 36)} onCardClick={clickOnCard} />
  ));
