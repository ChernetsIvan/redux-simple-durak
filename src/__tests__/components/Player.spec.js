import React from "react";
import Enzyme, { shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-15";
import toJson from "enzyme-to-json";

import DeckUtils from "./../../utils/DeckUtils";
import Player from "./../../components/GameScreen/Player/Player";

Enzyme.configure({ adapter: new Adapter() });

describe("Player shallow render test", () => {
  const cards = DeckUtils.fillCards({ suit: "Ð‘" });
  const emptyFunc = () => {};

  test("Player recieve > 10 cards (36)", () => {
    const component = shallow(<Player cards={cards} onCardClick={emptyFunc} />);
    expect(toJson(component)).toMatchSnapshot();
  });

  test("Player recieve < 10 cards (5)", () => {
    const component = shallow(<Player cards={cards.slice(0,5)} onCardClick={emptyFunc} />);
    expect(toJson(component)).toMatchSnapshot();
  });
});
