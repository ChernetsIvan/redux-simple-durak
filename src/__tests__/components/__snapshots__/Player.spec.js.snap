// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Player shallow render test Player recieve < 10 cards (5) 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col-10"
    >
      <div
        className="container"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-1 align-self-center"
          >
            <PrevButton
              cards={
                Array [
                  CardModel {
                    "id": 0,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 1,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 2,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 3,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 4,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                ]
              }
              onPrevClick={[Function]}
              startInd={0}
            />
          </div>
          <Card
            bootStrapColClass="col-1"
            id={0}
            key="Ч6"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 1,
                "text": "6",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={1}
            key="Ч7"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 2,
                "text": "7",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={2}
            key="Ч8"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 3,
                "text": "8",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={3}
            key="Ч9"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 4,
                "text": "9",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={4}
            key="Ч10"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 5,
                "text": "10",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <div
            className="col-1 align-self-center"
          >
            <NextButton
              cards={
                Array [
                  CardModel {
                    "id": 0,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 1,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 2,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 3,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 4,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                ]
              }
              endInd={9}
              onNextClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-auto"
    >
      <Connect(RemoveCardsButton) />
      <Connect(AiTakeButton) />
      <Connect(PlayerTakeButton) />
      <Connect(ShowStatisticsLink) />
    </div>
  </div>
</div>
`;

exports[`Player shallow render test Player recieve > 10 cards (36) 1`] = `
<div
  className="container"
>
  <div
    className="row"
  >
    <div
      className="col-1"
    />
    <div
      className="col-10"
    >
      <div
        className="container"
      >
        <div
          className="row justify-content-center"
        >
          <div
            className="col-1 align-self-center"
          >
            <PrevButton
              cards={
                Array [
                  CardModel {
                    "id": 0,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 1,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 2,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 3,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 4,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 5,
                    "power": 6,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 6,
                    "power": 7,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 7,
                    "power": 8,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 8,
                    "power": 9,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 9,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 10,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 11,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 12,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 13,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 14,
                    "power": 6,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 15,
                    "power": 7,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 16,
                    "power": 8,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 17,
                    "power": 9,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 18,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 19,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 20,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 21,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 22,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 23,
                    "power": 6,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 24,
                    "power": 7,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 25,
                    "power": 8,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 26,
                    "power": 9,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 27,
                    "power": 10,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 28,
                    "power": 20,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 29,
                    "power": 30,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 30,
                    "power": 40,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 31,
                    "power": 50,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 32,
                    "power": 60,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 33,
                    "power": 70,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 34,
                    "power": 80,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 35,
                    "power": 90,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                ]
              }
              onPrevClick={[Function]}
              startInd={0}
            />
          </div>
          <Card
            bootStrapColClass="col-1"
            id={0}
            key="Ч6"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 1,
                "text": "6",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={1}
            key="Ч7"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 2,
                "text": "7",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={2}
            key="Ч8"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 3,
                "text": "8",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={3}
            key="Ч9"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 4,
                "text": "9",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={4}
            key="Ч10"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 5,
                "text": "10",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={5}
            key="ЧВ"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 6,
                "text": "В",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={6}
            key="ЧД"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 7,
                "text": "Д",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={7}
            key="ЧК"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 8,
                "text": "К",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={8}
            key="ЧТ"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 9,
                "text": "Т",
              }
            }
            suit={
              Object {
                "suit": "Ч",
              }
            }
          />
          <Card
            bootStrapColClass="col-1"
            id={9}
            key="К6"
            onCardClick={[Function]}
            rank={
              Object {
                "cardValue": 1,
                "text": "6",
              }
            }
            suit={
              Object {
                "suit": "К",
              }
            }
          />
          <div
            className="col-1 align-self-center"
          >
            <NextButton
              cards={
                Array [
                  CardModel {
                    "id": 0,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 1,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 2,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 3,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 4,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 5,
                    "power": 6,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 6,
                    "power": 7,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 7,
                    "power": 8,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 8,
                    "power": 9,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "Ч",
                    },
                  },
                  CardModel {
                    "id": 9,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 10,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 11,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 12,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 13,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 14,
                    "power": 6,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 15,
                    "power": 7,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 16,
                    "power": 8,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 17,
                    "power": 9,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "К",
                    },
                  },
                  CardModel {
                    "id": 18,
                    "power": 1,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 19,
                    "power": 2,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 20,
                    "power": 3,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 21,
                    "power": 4,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 22,
                    "power": 5,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 23,
                    "power": 6,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 24,
                    "power": 7,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 25,
                    "power": 8,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 26,
                    "power": 9,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "П",
                    },
                  },
                  CardModel {
                    "id": 27,
                    "power": 10,
                    "rank": Object {
                      "cardValue": 1,
                      "text": "6",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 28,
                    "power": 20,
                    "rank": Object {
                      "cardValue": 2,
                      "text": "7",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 29,
                    "power": 30,
                    "rank": Object {
                      "cardValue": 3,
                      "text": "8",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 30,
                    "power": 40,
                    "rank": Object {
                      "cardValue": 4,
                      "text": "9",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 31,
                    "power": 50,
                    "rank": Object {
                      "cardValue": 5,
                      "text": "10",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 32,
                    "power": 60,
                    "rank": Object {
                      "cardValue": 6,
                      "text": "В",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 33,
                    "power": 70,
                    "rank": Object {
                      "cardValue": 7,
                      "text": "Д",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 34,
                    "power": 80,
                    "rank": Object {
                      "cardValue": 8,
                      "text": "К",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                  CardModel {
                    "id": 35,
                    "power": 90,
                    "rank": Object {
                      "cardValue": 9,
                      "text": "Т",
                    },
                    "suit": Object {
                      "suit": "Б",
                    },
                  },
                ]
              }
              endInd={9}
              onNextClick={[Function]}
            />
          </div>
        </div>
      </div>
    </div>
    <div
      className="col-1"
    />
  </div>
  <div
    className="row justify-content-center"
  >
    <div
      className="col-auto"
    >
      <Connect(RemoveCardsButton) />
      <Connect(AiTakeButton) />
      <Connect(PlayerTakeButton) />
      <Connect(ShowStatisticsLink) />
    </div>
  </div>
</div>
`;
